### PR TITLE
Deploying Directus to Microsoft Azure Web Apps

### LINKED ISSUE #40: Deploying Directus on Azure Cloud Service
### What is your idea?

Introduction:
This guide is designed to walk users through the process of deploying Directus, an open-source headless CMS, on the Azure Cloud Service for robust and scalable hosting. Starting with the creation of an Azure Web App, the tutorial will cover essential steps such as setting up an Azure Database for PostgreSQL, configuring environmental variables, and optimizing the deployment for seamless scalability.

Outline and Key Steps:
Creating an Azure Web App
Setting Up Azure Database for PostgreSQL
Deploying Directus on Azure Web App
Configuring Environment Variables
Optimizing for Scalability on Azure

Conclusion:
The guide will conclude by summarizing the deployment process, highlighting key considerations for security, scalability, and performance on Azure, and encouraging users to explore additional Azure features for enhanced Directus hosting.

### What are the key takeaways from your post?

The guide equips readers not only with the specific knowledge to deploy Directus on Azure but also imparts general skills in cloud-based application deployment, database configuration, security practices, and scalability principles applicable to a broader range of projects.

### What type of post is this?

Self-Hosting Guide

### Country of residence

Nigeria

### Terms & Conditions

- [X] You understand that submitting this form does not mean we accept your proposal. We will accept proposals at our discretion.
- [X] You understand that to be paid you will need to generate an invoice to us once the final draft of your work is completed.
- [X] You understand that you will grant us an unlimited license to post and promote your work as part of this program. You may repost, but using our copy as the canonical source.
- [x] You are aged 18 or over.

### PR DESCRIPTION
**Pull request by GeoSegun**
My complete submission

Closes #40

### CODE REVIEW

**Comment by GeoSegun** [comment]
Thanks, Kevin.

File: directus-az/index.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+![Azure Web App Basic Configuration](web-configuration.png)
+
+Click the **Next:Docker** button to configure the docker container settings for your web app using the docker-compose.yml we created earlier. Adjust the settings as seen below.
+
+![Docker congiguration settings](docker-configuration.png)
+
+Click the **Next:Networking** button and select your preferred network settings.
+Finally, click on **Review + Create** and then **Create** to create your new Web App
+
+Directus is successfully deployed and can be visited via the **Default domain** in the Azure web app. Leveraging the power of Directus' Backend as a Service (BaaS) solution, you can now benefit from ready-made REST and GraphQL APIs, seamless integrations, and superior scalability, empowering you to uncomplicate your backend operations with Directus.
```
Comment (phazonoverload) [inline directus-az/index.md:137]:
```suggestion
When creating a Web App, you will step through multiple configuration pages. Here are the important settings:

### Basics

- Subscription/Resource Group: select the same resource group we created and used earlier.
- Publish: Select "Docker Container". 

![Azure Web App Basic Configuration](web-configuration.png)

### Docker

Select Docker Compose and Docker Hub as the source for your app's configuration. Set the Docker Hub Access Type to Public and upload your `docker-compose.yml` file prepared earlier.

![Docker congiguration settings](docker-configuration.png)

Following the creation of the Web App Resource, Directus is now successfully deployed and can be visited via the default domain in the Azure Web App page. 
```

File: directus-az/index.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+## Trouble Shooting Tips
```
Comment (phazonoverload) [inline directus-az/index.md:139]:
```suggestion
## Troubleshooting Tips
```

File: directus-az/index.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+- **Admin Credentials:** Ensure that the `ADMIN_EMAIL` and `ADMIN_PASSWORD` values in your `docker-compose.yml` file match the credentials you are using to log in.
+
+- **Environment Variable Changes:** If you make changes to environment variables after the initial deployment, restart the Directus container to apply the new configurations.
+
+## Summary
+
+Deploying Directus on Azure Web App provides a robust solution for content management, offering scalability and flexibility for your web development projects. By leveraging Docker and Azure services like Azure Database for PostgreSQL, you create a reliable and efficient environment for managing databases and content.
+
+This tutorial has guided you through setting up a resource group, configuring Azure Database for PostgreSQL, and deploying Directus using Docker on Azure Web App. The integration of these technologies allows you to take advantage of the features offered by Directus while benefiting from the reliability and scalability of Azure services.
```
Comment (phazonoverload) [inline directus-az/index.md:172]:
```suggestion
This tutorial has guided you through setting up a resource group, configuring Azure Database for PostgreSQL, and deploying Directus using Docker on an Azure Web App.
```

**Review by phazonoverload — CHANGES_REQUESTED** [review]
Great post. A lot of my suggestions were about being more clear and direct, focusing on just the parts that matter to a Directus user deploying on Azure specifically.

**Comment by GeoSegun** [comment]
Thanks, Kevin.
I have successfully made the changes and also made a PR

**Pull request closed without merging**